lib/api/core/plugins/pluginContext.js

```diff
@@ -1,90 +1,124 @@
+    const Kuzzle = require('../../kuzzle');
+    let
+      pluginRepository,
+      pluginInternalEngine;
+
+    this.accessors = {};
+    this.config = kuzzle.config;
+    this.constructors = {
+      Dsl: require('../../dsl'),
+      Request: instantiateRequest,
+      RequestContext,
+      RequestInput,
+      BaseValidationType: require('../validation/baseType')
+    };
+    this.errors = require('kuzzle-common-objects').errors;
+
+    if (kuzzle instanceof Kuzzle) {
+      pluginRepository = new PluginRepository(kuzzle, pluginName);
+      pluginInternalEngine = new InternalEngine(kuzzle, `%${pluginName}`);
+      pluginInternalEngine.init(new PluginInternalEngineBootstrap(kuzzle, pluginInternalEngine));
+      pluginRepository.init({databaseEngine: pluginInternalEngine, cacheEngine: null});
+
+      Object.defineProperty(this, 'log', {
+        enumerable: true,
+        get: () => {
+          return {
+            silly: msg => kuzzle.pluginsManager.trigger('log:silly', msg),
+            verbose: msg => kuzzle.pluginsManager.trigger('log:verbose', msg),
+            info: msg => kuzzle.pluginsManager.trigger('log:info', msg),
+            debug: msg => kuzzle.pluginsManager.trigger('log:debug', msg),
+            warn: msg => kuzzle.pluginsManager.trigger('log:warn', msg),
+            error: msg => kuzzle.pluginsManager.trigger('log:error', msg)
+          };
+        }
+      });
+
+      Object.defineProperty(this.accessors, 'passport', {
+        enumerable: true,
+        get: () => {
+          return {
+            use: kuzzle.passport.use
+          };
+        }
+      });
 
-    Object.defineProperty(this.accessors, 'validation', {
-      enumerable: true,
-      get: () => {
-        return {
-          addType: kuzzle.validation.addType.bind(kuzzle.validation),
-          validate: kuzzle.validation.validationPromise.bind(kuzzle.validation)
-        };
-      }
-    });
+      Object.defineProperty(this.accessors, 'execute', {
+        enumerable: true,
+        get: () => {
+          return execute.bind(kuzzle);
+        }
+      });
+
+      /*
+       Sharing only a handful user functions as wrappers for the repositories object
+       */
+      Object.defineProperty(this.accessors, 'users', {
+        enumerable: true,
+        get: () => {
+          return {
+            load: userId => kuzzle.repositories.user.load(userId),
+            create: (name, profile, info) => createUser(kuzzle.repositories.user, name, profile, info)
+          };
+        }
+      });
+
+      Object.defineProperty(this.accessors, 'validation', {
+        enumerable: true,
+        get: () => {
+          return {
+            addType: kuzzle.validation.addType.bind(kuzzle.validation),
+            validate: kuzzle.validation.validationPromise.bind(kuzzle.validation)
+          };
+        }
+      });
+
+      Object.defineProperty(this.accessors, 'storage', {
+        enumerable: true,
+        get: () => {
```

Comment:
If I understand properly, each time the `accessors.storage` getter is called, it returns a new object, which I am not sure is the desired behaviour.

I would rather expose a constructor, that would force the plugin index but let plugin creators to create & use multiple collections.